app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,31 @@
+class CommentsController < ApplicationController
+  before_action :set_user_and_post
+
+  def create
+    @comment = @post.comments.new(comment_params)
+    @comment.user = current_user # assuming you have a current_user method
+
+    if @comment.save
+      # Update the comments list and comments_counter
+      @comments = @post.comments.order(created_at: :desc)
+      @post.update(comments_counter: @comments.count)
+
+      # Redirect or render as needed
+      redirect_to user_post_path(@user, @post), notice: 'Comment added successfully'
+    else
+      # Handle validation errors
+      render 'posts/show'
+    end
+  end
```

Comment:
- [ ] Great job so far in your implementation of the forms on your blog app. However, I noticed a few issues that when adjusted would help improve the user experience and functionalities of the app.
   - I noticed you have your comment form placed in the post details page which looks okay but doesn't match project requirements. Your comment form should have a dedicated route(`/posts/:post_id/comments/new`) just like you did for the post form.
   
   - Also, when creating a comment it throws an error that prevents users from commenting on posts. Below is a full details of the error in a screenshot.
   
<img width="1016" alt="Screenshot 2024-01-20 at 21 59 27" src="https://github.com/Eccb7/Pi-Blogs/assets/107192076/ac38e23d-e8e4-4dd1-b82c-56aedf324d7c">